### PR TITLE
Enable EqualsAvoidsNull (and others causing no effect)

### PR DESCRIPTION
**Pull request by koppor**
Adds https://docs.openrewrite.org/recipes/java/cleanup/equalsavoidsnull

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Review by Siedlerchr — APPROVED** [review]

File: src/test/java/org/jabref/logic/importer/fetcher/SpringerFetcherTest.java [hunk 1]
```diff
@@ -182,9 +182,9 @@ public void supportsBooleanANDSearch() throws Exception {
 
         Assertions.assertTrue(resultJustByAuthor.containsAll(result));
         List<BibEntry> allEntriesFromCSCW = result.stream()
-                                                  .filter(bibEntry -> bibEntry.getField(StandardField.JOURNAL)
-                                                                              .orElse("")
-                                                                              .equals("Computer Supported Cooperative Work (CSCW)"))
+                                                  .filter(bibEntry -> "Computer Supported Cooperative Work (CSCW)"
+                                                                              .equals(bibEntry.getField(StandardField.JOURNAL)
+                                                                              .orElse("")))
```
Comment (calixtus) [inline src/test/java/org/jabref/logic/importer/fetcher/SpringerFetcherTest.java:187]: This seems to be NPE safe before, as the Optional returns either the contents of the field or an empty string. In either case equals wont be run on null.

Comment (koppor) [reply src/test/java/org/jabref/logic/importer/fetcher/SpringerFetcherTest.java:187]: These are the rules 🤷‍♂️. Should we disable the rule and only commit the changes manually? 😅

Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/importer/fetcher/SpringerFetcherTest.java:187]: I think this is an edge case here, but in general the rule is useful

**Review by Siedlerchr — APPROVED** [review]

File: src/test/java/org/jabref/logic/importer/fetcher/SpringerFetcherTest.java (hunk 1 above)
Comment (calixtus) [reply src/test/java/org/jabref/logic/importer/fetcher/SpringerFetcherTest.java:187]: So there seems to be a problem with the rules, unable to understand optionals. Let's talk in the devcall about pros and cons

Comment (koppor) [reply src/test/java/org/jabref/logic/importer/fetcher/SpringerFetcherTest.java:187]: I filed https://github.com/openrewrite/rewrite-static-analysis/issues/97 - think, we should close this pull request.

I also find the rewrites not intuitive. Let's discuss.

**Review by calixtus — APPROVED** [review]

**Pull request merged**